crypto/crypto/src/x25519.rs

```diff
@@ -206,6 +206,12 @@ impl ExchangeKey for X25519StaticPrivateKey {
     }
 }
 
+impl PartialEq for X25519StaticPrivateKey {
+    fn eq(&self, other: &Self) -> bool {
+        self.to_bytes() == other.to_bytes()
+    }
+}
+
```

Comment:
I think there's a good reason this wasn't implemented before @kchalkias and @huitseeker for input.

Reply:
Meeting with one of them tomorrow to gather more general feedback on the Crypto API. There's a lot of non-trivial logic around this space to handle keys, which I'd love to simplify as much as possible.

Reply:
As long as `to_bytes()` is exposed, I see no reason to not support `PartialEq`. It's fine to implement it; we already allow it in [ed25519](https://github.com/davidiw/libra/blob/d5e277853939ccc5eef16dde4aa47bc4b5b8bdcb/crypto/crypto/src/ed25519.rs#L178).

Reply:
Then why all the concern over not implementing `Clone`?

Reply:
Clone is different, you're copying the bytes somewhere else. This is just about comparing, nothing harmful should happen by mistake here.

Reply:
Perhaps in our current state of the world this was an over optimization? I think the ideal case would be to pin this data to the heap: https://doc.rust-lang.org/std/sync/struct.Arc.html#method.pin

I personally would like to eliminate the current attempts to minimize / the movement of the key and at a later date follow up with using cleaner semantics that don't require special containers for moving private keys from disk to config to usage (that may be littering memory with the key) and  invariants in the code base (such as do not clone a private key).

Who would be good folks to talk to with respect to simplifying this?

Reply:
The idea was that all of this would be temporary btw, as we would move all keys and key operations to a secret service. There were some people involved in this, I'm not sure who is leading this anymore. @valerini @weilei0107 @gedigi ?

Reply:
But the fact that there exists a `to_bytes()` method means you can Clone the key without calling a function called `clone()`.